### ISSUE TITLE
CUDA warnings – `long double` is treated as `double`

### ISSUE THREAD

**Issue by sayerhs** [Exawind/nalu-wind#360]
We recently get these warnings when building Nalu-Wind with CUDA 

```
../src/kernel/ContinuityAdvElemKernel.C(91): warning: 'long double' is treated as 'double' in device code
```

These are appearing due to changes introduced in #315 

https://github.com/Exawind/nalu-wind/blob/626b95fc5c3c61ed8f3c7c32c0da5f8e772d350a/include/KokkosInterface.h#L17-L21

on lines such as these

https://github.com/Exawind/nalu-wind/blob/626b95fc5c3c61ed8f3c7c32c0da5f8e772d350a/src/kernel/ContinuityAdvElemKernel.C#L91-L94

@overfelt Should we go back to your `max_align_t` suggestion? or should we just use `double` on CUDA builds?

**Comment by alanw0** [comment]
These alignment issues have been a problem off and on for a long time. alignas(double) sounds good to me for cuda...

**Comment by sayerhs** [comment]
Fixed in #364
